artcommon/artcommonlib/konflux/konflux_db.py

```diff
@@ -339,18 +343,29 @@ async def get_build_record_by_nvr(self, nvr: str, outcome: KonfluxBuildOutcome =
         self.logger.warning('No builds found for NVR %s', nvr)
         return None
 
-    async def get_build_records_by_nvrs(self, nvrs: typing.Sequence[str], outcome: KonfluxBuildOutcome = KonfluxBuildOutcome.SUCCESS, strict: bool = True) -> typing.List[KonfluxRecord]:
+    async def get_build_records_by_nvrs(self, nvrs: typing.Sequence[str],
+                                        outcome: KonfluxBuildOutcome = KonfluxBuildOutcome.SUCCESS,
+                                        where: typing.Optional[typing.Dict[str, typing.Any]] = None,
+                                        strict: bool = True) -> typing.List[KonfluxRecord]:
         """ Get build records by NVRS.
         Note that this function only searches for the build records in the last 3 years.
 
         :param nvrs: The NVRS of the builds.
         :param outcome: The outcome of the builds.
+        :param where: Additional fields to filter the build records.
+        :param strict: If True, raise an exception if any build record is not found.
         :return: The build records.
         """
         nvrs = list(nvrs)
+        if not where:
+            where = {}
+        else:
+            if "nvr" in where or "outcome" in where:
+                raise ValueError("'nvr' and 'outcome' fields are reserved and should not be used in the 'where' "
+                                 "parameter")
 
         async def _task(nvr):
-            where = {"nvr": nvr, "outcome": str(outcome)}
+            where.update({"nvr": nvr, "outcome": str(outcome)})
```

Comment:
If non-empty where dict is passed from the parameter, this will modify the dict and will persist across subsequent method invocations. Also if we run multiple queries at the same time (by passing more than one NVR), this may cause a data race.

A safer way is to where.copy().update(...) so that the original dict will never be modified.

Reply:
Good point! will change this

Reply:
Done